00-open-a-shapefile.Rmd

```diff
@@ -78,237 +75,254 @@ Working With Shapefile Attributes in R </a>
+includes the _class_, the number of _features_, the _extent_, and the coordinate
+reference system (`CRS`) of the `R` object. The last two lines of summary show a
+preview of the `R` object _attributes_.
 
-#Import a line and point shapefile
-
+```{r shapefile-summary}
+#view a summary of metadata & attributes associated with the spatial object
+summary(aoiBoundary_HARV)
 
->#Challenge: Import a line and point shapefile
->
->Using the steps above, import the HARV_roads and HARVtower_UTM18N layers into R.
->Call the Harv_roads object `lines` and the HARVtower_UTM18N `point`
->Answer the following questions:
+```
 
->1. What type of R spatial object is created when you import each layer?
->2. What is the extent and CRS for each object?
->3. Do the files contain, points, lines or polygons?
->4. How many spatial objects are in each file?
->
->Discuss the data with your neighbor as you work through this challenge!
+##Challenge: Attributes & Metadata
+What is the difference between the types of data you get from just entering the
+object name, `summary()` and `attributes()`. When might you prefer to use one 
+or the other?
 
+``` {r challenge-code-attributes, echo=FALSE}
+# Entering just the name provides the most commonly needed information: CRS,
+# Extent, item names, and min & max values.  `attributes()` prints out all the 
+# "slots" associated with the object.  These "slots" are areas where different
+# types of data can be stored. Notice that `attributes()` provides all the
+# information the name alone does while providing more information as well. 
+# Summary provides 
 
-```{r import-point-line, echo=FALSE, results="hide" }
+```
 
-#Import a line shapefile
-lines <- readOGR("boundaryFiles/HARV",layer = "HARV_roads")
-#import a point shapefile
-point <- readOGR("boundaryFiles/HARV", layer="HARVtower_UTM18N")
-  
+#Plot a Shapefile
+Now that we have become familiar with our object attributes, we can visualize
+our data. Using `plot`, we can produce a box representing the boundaries
+specified in this shapefile.
 
-#view attributes
-lines
-point
+``` {r plot-shapefile}
+#create a plot of the shapefile
+#'lwd' sets the line width. 'col' sets internal color, 'border' sets line color
+plot(aoiBoundary_HARV, col="cyan1", border="black", lwd=3)
 
 ```
 
-#Shapefile Metadata & Attributes
+#Challenge: Import Line and Point Shapefiles
+Using the steps above, import the HARV_roads and HARVtower_UTM18N layers into
+`R`. Call the Harv_roads object `lines_HARV` and the HARVtower_UTM18N
+`point_HARV`.
 
-In addition to the metadata, each object has one or more attributes linked 
-to it.  Attributes of an object can include measurements, descriptions, or
-classifications of the object. Attribute data can be any sort of information that you might 
-want to associate with your spatial objects. Each spatial 
-object in a shapefile can have the same set of attributes or each can be 
-different. 
+Answer the following questions:
 
```

Comment:
The answers are in the code below the challenge.  For ones that were just the output of the code I didn't re-write the number (e.g. extent)  but included the code.